### PR TITLE
Add questions and notes for pr20827

### PR DESCRIPTION
**Pull request by 0xB10C**
Frohe Weihnachten :cactus:

### CODE REVIEW

File: _posts/2021-12-29-#20827.md [hunk 1]
```diff
@@ -10,15 +10,54 @@ status: upcoming
...
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+- When pruning is enabled in Bitcoin Core, data about old blocks is deleted to limit disk space usage.
+Users can configure a pruning target defining how much disk space to use for block and undo data.
+The minimum target is 550MB.
```
Comment (glozow) [inline _posts/2021-12-29-#20827.md:17]: Suggestion - mention and/or link to `-prune` option

File: _posts/2021-12-29-#20827.md [hunk 2]
```diff
@@ -10,15 +10,54 @@ status: upcoming
...
+- PR [#12404](https://github.com/bitcoin/bitcoin/pull/12404) attempted aggressive pruning too, but was closed in favor of PR [#11658](https://github.com/bitcoin/bitcoin/pull/11658).
+PR #11658 added 10% of the prune target to the `nBuffer`.
+This is being overwritten by this PR.
+
+
 ## Questions
 1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
--->
 
+2. Where in the code do we check if we need to prune old block data?
```
Comment (glozow) [inline _posts/2021-12-29-#20827.md:50]: Perhaps add a hint for what to grep for?

File: _posts/2021-12-29-#20827.md [hunk 3]
```diff
@@ -10,15 +10,54 @@ status: upcoming
...
+
+3. What is removed during pruning and under which conditions? What is not pruned?
+
+4. What is the variable `nBuffer` in [`BlockManager:FindFilesToPrune()`](https://github.com/bitcoin/bitcoin/blob/24f3936337de3afb4fa56efc83009e2527d22df0/src/validation.cpp#L3622) being used for? How large is the buffer (in MB)?
+
+5. The PR assumes 1MB for `average_block_size`. How accurate does this assumption have to be?
+
+6. The PR description mentions IBD speed improvements for pruned nodes. Which metrics can we measure to benchmark the improvement? With which prune targets and dbcache sizes should we test?
```
Comment (glozow) [inline _posts/2021-12-29-#20827.md:58]:
```suggestion
6. The PR description mentions IBD speed improvements for pruned nodes. What can we measure to benchmark the improvement? With which prune targets and dbcache sizes should we test?
```

File: _posts/2021-12-29-#20827.md [hunk 4]
```diff
@@ -10,15 +10,54 @@ status: upcoming
...
+
+7. Edge case: Is agressivly pruning during IBD a problem in-case there are longer forks in the chain?
```
Comment (glozow) [inline _posts/2021-12-29-#20827.md:60]:
```suggestion
7. Edge case: Is agressively pruning during IBD a problem if there are longer forks in the chain?
```

File: _posts/2021-12-29-#20827.md [hunk 5]
```diff
@@ -10,15 +10,54 @@ status: upcoming
...
+We now aggressively prune all prunable files enabling us to continue with IBD without having to prune again too soon.
+Fewer prunes also mean fewer dbcache flushes, potentially speeding IBD for pruned nodes up.
+Higher dbcache sizes can be reached before the dbcache is flushed.
+
+<p style="text-align:center"><a href="../assets/img/20827-pruning.drawio.png"><img src="../assets/img/20827-pruning.drawio.png" style="max-width:100%; height:auto"></a></p>
+
+
+- PR [#12404](https://github.com/bitcoin/bitcoin/pull/12404) attempted aggressive pruning too, but was closed in favor of PR [#11658](https://github.com/bitcoin/bitcoin/pull/11658).
+PR #11658 added 10% of the prune target to the `nBuffer`.
+This is being overwritten by this PR.
```
Comment (glozow) [inline _posts/2021-12-29-#20827.md:44]:
```suggestion
This is being overwritten by PR #20827.
```

File: _posts/2021-12-29-#20827.md [hunk 6]
```diff
@@ -10,15 +10,54 @@ status: upcoming
...
+
+
 ## Questions
 1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
```
Comment (glozow) [inline _posts/2021-12-29-#20827.md:48]: Doesn't need to be a question here, but I recommend asking "what does this PR do" during the meeting, especially since the diff is so small.

**Review by glozow — COMMENTED** [review]
Thanks so much! Nice diagram :)

Frohe Weihnachten! 🎄

**Comment by 0xB10C** [comment]
Thank you! I found the suggestions helpful.

**Comment by glozow** [comment]
Thanks @0xB10C!
